yt_dlp/extractor/vanillo.py

```diff
@@ -0,0 +1,386 @@
+                'text': reply.get('text'),
+                'timestamp': parse_iso8601(reply.get('createdAt')),
+            }
+            replies.append(transformed)
+        return replies
+
+    def _get_comments(self, video_id, limit=10):
+        all_comments = []
+        page_key = None
+        # Loop to download all comments using pageKey
+        while True:
+            url = (
+                f'https://api.vanillo.tv/v1/videos/{video_id}/comments'
+                f'?limit={limit}&reviewing=false&filter=high_to_low_score'
+            )
+            if page_key:
+                url += f'&pageKey={page_key}'
+            try:
+                comments_data = self._download_json(url, video_id, note='Downloading comments', fatal=False)
+            except ExtractorError:
+                break
+            if comments_data.get('status') != 'success':
+                break
+            data = comments_data.get('data', {})
+            comments = data.get('comments', [])
+            if not comments:
+                break
+            # For each comment, download replies (if any)
+            for comment in comments:
+                transformed = {
+                    'id': comment.get('id'),
+                    'author': comment.get('profile', {}).get('username'),
+                    'author_id': comment.get('profile', {}).get('id'),
+                    'text': comment.get('text'),
+                    'timestamp': parse_iso8601(comment.get('createdAt')),
+                    'replies': self._get_replies(comment.get('id')),
+                }
+                all_comments.append(transformed)
+            page_key = data.get('nextPageKey')
+            if not page_key:
+                break
+        return all_comments
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        # 1) Retrieve video info (metadata)
+        video_info_url = f'https://api.vanillo.tv/v1/videos/{video_id}?groups=uploader,profile.full'
+        try:
+            video_info = self._download_json(video_info_url, video_id, note='Downloading video info')
+        except ExtractorError as e:
+            # Try to get an HTTP code from the error cause or message
+            http_code = getattr(e.cause, 'code', None)
+            if http_code is None and 'HTTP Error 404' in str(e):
+                http_code = 404
+            if http_code == 404:
+                self.raise_login_required(
+                    'Video not found or is private. '  # maybe use report_login here?
+                    'Sign in if you have access to this video. Use --cookies or --cookies-from-browser option',
+                    method=None,
+                )
+            elif http_code == 403:
+                raise ExtractorError(
+                    'Your Internet provider is likely blocked. Try again with another ISP or use VPN',
+                    expected=True,
+                )
+            raise
```

Comment:
add this to imports: `from ..networking.exceptions import HTTPError`

then:
```suggestion
        except ExtractorError as e:
            if not isinstance(e.cause, HTTPError):
                raise
            if e.cause.status == 404:
                self.raise_login_required('Video not found or is private')
            elif e.cause.status == 403:
                raise ExtractorError(
                    'Your IP address is likely blocked. Try from another IP', expected=True)
            raise
```